mlflow/genai/git_versioning/__init__.py

```diff
@@ -0,0 +1,59 @@
+import warnings
+
+from typing_extensions import Self
+
+from mlflow.genai.git_versioning.git_info import GitInfo, GitOperationError
+from mlflow.utils.annotations import experimental
+
+
+class GitContext:
+    def __init__(self) -> None:
+        try:
+            self.info = GitInfo.from_env()
+        except GitOperationError as e:
+            warnings.warn(
+                f"Git operation failed: {e}",
+                UserWarning,
+                stacklevel=2,
+            )
+            self.info = None
```

Comment:
Will this warning print if I'm not in a git repo?

Reply:
Yes, it does. `test_disable_git_model_versioning_in_non_git_repo` is a test